data/economy/commodities/agricultural.json

```diff
@@ -5,37 +5,37 @@
 			"inputs": [ "farm_machinery", "fertilizer" ],
 			"consumable": true,
 			"producer": "agricultural",
-			"price": 105
+			"price": 675
```

Comment:
As you very well understood, the problem on current state is that all, but start ships, have a constant maximum profit, by trading the most expensive commodity, _whatever is his name or price._
By accepting the exponential in 'rn', ship's will have steady, but different on each ship, maximum profit by trading some, particular to each ship, ideal commodity defined by its ideal price. You can look at my_economy sheet in [commodities.ods](https://github.com/jimishol/Pioneerspacesim_trading_economy/blob/main/docs/commodities.ods) for some example. Skipjack will have his maximum profit only if he trades a commodity that major exports in 67.528t(cell L23). That means if it finds a commodity with a price that equals to Calibration.BestCommodityPrice(67.528)=1159.7133520942. _I used print (require  "Calibration".BestCommodityPrice(67.528)) from lua terminal._ If such a commodity do exist, all are good. If it does not, ship will earn less and will have difficulties to find its rightful place.

Prices on existing commodities come to give meaning on existing ships. They are irrelevant to the exponential and so, we can safely remove this part and reject alterations on commodities prices without harming the scope of present PR. Doing so, ships will get less than their potential best profit, because, to some, there will be no commodities with prices equal to ship's ideal ones.

The most important role of the exponential is that it defines an ideal commodity price for each ship. Upon an agreement on the 'global', let me characterize it as such, value of exponential, either developers will alter the price of some commodity to fulfill the ideal profit of some ship or ship designer will build a ship such that its ideal commodity price will be equal or close enough to some existed commodity. Ideal commodity prices of ships will always exist as parts of the ships. Existed commodities can be altered or even deleted. Think that current state is equivalent to a state where all commodities, but precious metals and fuel, are deleted from the game. Pioneer would still operate pretty much as it does today.

Alteration of commodity prices is not important in this PR. They just demonstrate a working example of the consequences of the existence of an 'rn's exponential. In that scope they can safely rejected.
if my English hadn't betrayed me and was clear enough, it is fine by me to 'resolve conversation' of this one.

Reply:
Most of these conversations can indeed be considered as "resolved" regardless of whether they are marked as such, as they're not really requests for change; I just wanted to clearly manage expectations for what would be merged and what might require further thought or will most likely not be merged due to conflicts with the vision of the game. :smile:

One thing I wanted to bring up is that your response highlights the idea of an ideal / "specialization" commodity for each given ship, and your approach of tweaking the commodities to meet the ships and vice versa. Something I'd actually like to do is to attempt to allow the player to have multiple choices where possible, so that there is more than one commodity that they can trade for near-maximum profit in a specific ship. Certainly, the nature of commodities and profits means that there will eventually be a maxima of the "preferred commodity" for a given ship, but where possible I'm happy to see a ship that sits in the gap between two or three different commodities where the profit for each commodity is within a small margin of the average profit of them all, as this allows the player more options when choosing a trade route.

Reply:
I played my mod many hours. I wish you could find some time,  as a player, to play it too, to see it in action. I guaranty that in practice it works just as you like. Start a new game, acquire some middle ship and try to maximize your profit. Because of huge variations on stock, along side with major minor mode's alterations in price, player has very hard time to find his ideal commodity. I do not cheat and do not look at ideal prices of my ship. I play hours and i am not sure about my perfect commodity. By looking at my available cargo and the stock i trade, i am sure that i am not on best commodity but i always wonder what might be that. Once i find it _(many times I feel that i changed ships without finding the ideal commodity of my previous ship)_, the drop in offer is detectable and I start search for a new system. The not existence of major or minor exports on some commodities makes that even harder. As it is, _both large variations and luck of exports benefits the challenge of the game and should not be altered._ If you narrow them and remove the luck of exports, i believe that the math behind all of this will be visible and boring and your fears in your second paragraph about the "prefered commodity"will be more than true.

Inspired by @nozmajner 's comment about negotiations, that i liked a lot, and your last reply, I would like the following. By exponential, my profit functions will work by themselves, no matter if they are used by code or not. A stock and price will appear at player to buy as usual. The same could be applied to mission rewards too. A bill could be created by player. The player will see the usual price stock values but before he actually buy or sell, an auto created bill, supposed from player, would be present to station. That would say "current system price * stock = invest, + or - invest x ECM%, + or - invest x atmo shield %, plus or minus  an % ratio of invest for what ever equipment or reason developers like and agree upon them - Sum  = this" Acceptance by station could be based on popularity or whatever. In that way, the exponential's unavoidable consequence to define, behind developers' back, an ideal for each ships commodity price, is %altered my the equipment creations of developers and player's choices to pick them or not.